k8s-pb-codegen/src/main.rs

```diff
@@ -95,5 +105,172 @@ fn main() -> Result<()> {
         std::fs::rename(src, dst)?;
     }
 
+    let mut lib_rs = OpenOptions::new()
+        .append(true)
+        .open(target_dir.join("lib.rs"))
+        .unwrap();
+    append_trait_def(&mut lib_rs);
+
     Ok(())
 }
+
+fn append_trait_def(lib_rs: &mut File) {
+    let tokens = quote! {
+        pub trait Resource {
+            const API_VERSION: &'static str;
+            const GROUP: &'static str;
+            const KIND: &'static str;
+            const VERSION: &'static str;
```

Comment:
Ugh, GKV. This bothers me, so let me fix and regenerate. I'll merge this after that.